flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/datasource/statements/JdbcQueryStatement.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.jdbc.datasource.statements;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import java.io.Serializable;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+/**
+ * Sets {@link PreparedStatement} parameters to use in JDBC Sink based on a specific type of record.
+ */
+@PublicEvolving
+public interface JdbcQueryStatement<T> extends Serializable {
+    String query();
+
+    void map(PreparedStatement ps, T data) throws SQLException;
```

Comment:
- `data` -> `record`
- The method described by the interface here is more like filling the data of a message into a preparedstatement. So how about    
     - changing method name like `JdbcStatementFiller` or `JdbcStatementRender`
     - changing the interface name to `fill` or `render`     
?

Reply:
This is to have the query, and how to set values to the PStatement on same place..
calling JdbcStatementFiller for example we are focus on the second method..

I agree we can change map to another thing.